pkg/serializer/split/split.go

```diff
@@ -45,19 +45,19 @@ func init() {
 
 // CheckSizeAndSerialize Check the size of a payload and marshall it (optionally compress it)
 // The dual role makes sense as you will never serialize without checking the size of the payload
-func CheckSizeAndSerialize(m marshaler.Marshaler, compress bool, mType MarshalType) (bool, []byte, error) {
-	payload, _, err := serializeMarshaller(m, compress, mType)
+func CheckSizeAndSerialize(m marshaler.Marshaler, compress bool, mType MarshalType) (bool, []byte, []byte, error) {
```

Comment:
It technically works but I think the signature change has a small problem here:

There is a flag to compress _or not_ but you always return two payloads (compressed/raw) and if `compress == true`, you return both time the raw payload.
Also, if we look at your 2 usages while asking for a compressed payload:
 - you're discarding the uncompressed one (split.go:106)
 - you're only using the uncompressed payload for the debug message (serializer.go:261)

It works, but we maybe need a helper method here or a small refactor?

Reply:
yes that's fair, went for the smallest change set but clearly not the cleanest. I'll see what I can do to clean this up.

Reply:
I think there are fundamental problems in the signature of these functions, I went down the way of refactoring them but I can't find a way to do so while keeping the change set small.

Also, logging the uncompressed payload is critical for debugging.

If you see an obvious small improvement to make here let me know, otherwise I'll go ahead and merge this (and make a card to refactor the logic to serialize and compress metadata payloads, which shouldn't be handled differently compared to other payload types)

Reply:
Nothing obvious in mind, go ahead 👍

Reply:
thanks 🙇